### PR TITLE
[DOCS] Adds dataframe authorization details

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds authorization details for the data frame APIs.

It also enables testing for the examples in the create data frame transform API.

### CODE REVIEW

**Comment by lcawl** [comment]
You can preview these changes in this section of the docs:
https://doc-review-63bf6.firebaseapp.com/es5/data-frame-apis.html

File: docs/reference/data-frames/apis/get-transform-stats.asciidoc [hunk 1]
```diff
@@ -42,7 +44,11 @@ The API returns the following information:
 `transforms`::
   (array) An array of statistics objects for {dataframe-transforms}.
 
-//==== Authorization
+==== Authorization
+
+If the {es} {security-features} are enabled, you must have
+`data_frame_transforms_user` cluster privileges to use this API. For more
```
Comment (benwtrent) [inline docs/reference/data-frames/apis/get-transform-stats.asciidoc]: The specific cluster privilege is `monitor_data_frame_transforms`. A role that has that privilege is `data_frame_transforms_user`

File: docs/reference/data-frames/apis/delete-transform.asciidoc [hunk 2]
```diff
@@ -22,7 +24,11 @@ Deletes an existing {dataframe-transform}.
 `data_frame_transform_id` (required)::
   (string) Identifier for the {dataframe-transform}.
 
-//===== Authorization
+==== Authorization
+
+If the {es} {security-features} are enabled, you must have
+`manage_data_frame_transforms` cluster privileges to use this API.
```
Comment (benwtrent) [inline docs/reference/data-frames/apis/delete-transform.asciidoc]: The cluster privilege is `manage_data_frame_transforms` a role that has that privilege is `data_frame_transforms_admin`. 

I am not sure if we document the specific privileges or the roles we created ourselves.

File: docs/reference/data-frames/apis/get-transform.asciidoc [hunk 3]
```diff
@@ -53,7 +55,11 @@ The API returns the following information:
 `transforms`::
   (array) An array of transform resources.
 
-//==== Authorization
+==== Authorization
+
+If the {es} {security-features} are enabled, you must have
+`data_frame_transforms_user` cluster privileges to use this API. For more
```
Comment (benwtrent) [inline docs/reference/data-frames/apis/get-transform.asciidoc]: The specific cluster privilege is `monitor_data_frame_transforms`. A role that has that privilege is `data_frame_transforms_user`

File: docs/reference/data-frames/apis/preview-transform.asciidoc [hunk 4]
```diff
@@ -26,6 +27,13 @@ Previews a {dataframe-transform}.
 `pivot`:: Defines the pivot function `group by` fields and the aggregation to
 reduce the data.
 
+==== Authorization
+
+If the {es} {security-features} are enabled, you must have
+`manage_data_frame_transforms` cluster privileges to use this API. You must also
+have `view_index_metadata` privileges on the source index for the
+{dataframe-transform}. For more information, see
```
Comment (benwtrent) [inline docs/reference/data-frames/apis/preview-transform.asciidoc]: `view_index_metadata` and `read` privileges on the source index.

File: docs/reference/data-frames/apis/put-transform.asciidoc [hunk 5]
```diff
@@ -44,7 +46,13 @@ reduce the data. See <<data-frame-transform-pivot, data frame transform pivot ob
 `description`:: Optional free text description of the data frame transform
 
 
-//==== Authorization
+==== Authorization
+
+If the {es} {security-features} are enabled, you must have
+`manage_data_frame_transforms` cluster privileges to use this API. You must also
+have `read` privileges on the source index and `read`, `create_index`, and
```
Comment (benwtrent) [inline docs/reference/data-frames/apis/put-transform.asciidoc]: `view_index_metadata` for the source index

File: docs/reference/data-frames/apis/preview-transform.asciidoc (hunk 4 above)
Comment (lcawl) [reply docs/reference/data-frames/apis/preview-transform.asciidoc]: Thanks, I've added that privilege!

File: docs/reference/data-frames/apis/delete-transform.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/reference/data-frames/apis/delete-transform.asciidoc]: We didn't put the built-in roles in the ML APIs, but I think it's helpful so I've added it here

**Review by benwtrent — APPROVED** [review]

**Pull request merged**